File: C#/Projects/SnomedQuery.Parser/RF2Concept.cs
﻿using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SnomedQuery.Parser
{
  /// <summary>
  /// Stored data from one line of the RFS concept file.
  /// </summary>
  public class RF2Concept : RF2ItemSingle
  {
    /// <summary>
    /// Snomed definition status concept id.
    /// </summary>
    public Int64 DefinitionStatusId;

    /// <summary>
    /// Constructor.
    /// </summary>
    public RF2Concept()
    {
    }

    /// <summary>
    /// Parse a single line of snomed concept rf2 file.
    /// </summary>
    /// <param name="parser"></param>
    /// <param name="line"></param>
    /// <returns></returns>
    public static RF2Concept Parse(RF2Parser parser, String line)
    {
      String[] parts = line.Split('\t');
      if (parts.Length != 5)
        throw new ApplicationException("Invalid concept line");

      return new RF2Concept()
      {
        Parser = parser,
        Id = Int64.Parse(parts[0]),
        EffectiveTime = ParseEffectiveTime(parts[1]),
        Active = ParseBool(parts[2]),
        ModuleId = Int64.Parse(parts[3]),
        DefinitionStatusId = Int64.Parse(parts[4])
      };
    }

    /// <summary>
    /// ToString() overload.
    /// </summary>
    /// <returns></returns>
    public override string ToString()
    {
      return $"RF2Concept [{Id}] Active: {Active} Module: {ModuleId}";
    }
  }
}

File: Java/SnomedQuery/Parser/RF2Concept.java
package Parser;

/**
 * <p>
 * Stored data from one line of the RFS concept file.
 * </p>
 * 
 * @author Travis Lukach
 */
public class RF2Concept extends RF2ItemSingle {
	/**
	 * <p>
	 * Snomed definition status concept id.
	 * </p>
	 */
	private long definitionStatusId;

	/**
	 * <p>
	 * gets <i>definitionStatusId</i>.
	 * </p>
	 * 
	 * @return
	 */
	public long getDefinitionStatusId() {
		return definitionStatusId;
	}

	/**
	 * sets <i>definitionStatusId</i>.
	 * @param definitionStatusId
	 */
	public void setDefinitionStatusId(long definitionStatusId) {
		this.definitionStatusId = definitionStatusId;
	}
	/**
	 * <p>
	 * Constructor.
	 * </p>
	 */
	public RF2Concept() {
	}

	/**
	 * Parse a single line of snomed concept rf2 file.
	 * 
	 * @param parser
	 * @param line
	 * @return
	 */
	public static RF2Concept parse(RF2Parser parser, String line) {
		String[] parts = line.split("[\\t]", -1);
		if (parts.length != 5) {
			throw new RuntimeException("Invalid concept line");
		}

		RF2Concept tempVar = new RF2Concept();
		tempVar.setParser(parser);
		tempVar.setId(Long.parseLong(parts[0]));
		tempVar.setEffectiveTime(parseEffectiveTime(parts[1]));
		tempVar.setActive(parseBool(parts[2]));
		tempVar.setModuleId(Long.parseLong(parts[3]));
		tempVar.setDefinitionStatusId(Long.parseLong(parts[4]));
		return tempVar;
	}

	/**
	 * ToString() overload.
	 * 
	 * @return
	 */
	@Override
	public String toString() {
		return String.format("RF2Concept [%1$s] Active: %2$s Module: %3$s", getId(),
				isActive(), getModuleId());
	}

}